libsolidity/interface/CompilerStack.cpp

```diff
@@ -841,3 +842,81 @@ string CompilerStack::computeSourceMapping(eth::AssemblyItems const& _items) con
 	}
 	return ret;
 }
+
+namespace {
+
+Json::Value gasToJson(GasEstimator::GasConsumption const& _gas)
+{
+    if (_gas.isInfinite || _gas.value > std::numeric_limits<Json::LargestUInt>::max())
+	return Json::Value(Json::nullValue);
```

Comment:
Alternatively we could make it a string and then no truncation is required. Infinity is then either a string as well or a special value (such as -1?)